tests/markov/msm/test_msm.py

```diff
@@ -242,797 +220,485 @@ def test_valid_trajectory(self):
+
+
+def test_discrete_trajectories_active(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    dta = msm.count_model.transform_discrete_trajectories_to_submodel(scenario.dtraj)
+    assert_equal(len(dta), 1)
+    # HERE: states are shifted down from the beginning, because early states are missing
+    assert_(dta[0][0] < scenario.dtraj[0])
+
+
+def test_physical_time(double_well_msm_all):
+    _, _, msm = double_well_msm_all
+    assert_(str(msm.count_model.physical_time).startswith('1'))
+    assert_(str(msm.count_model.physical_time).endswith('step'))
+
+
+def test_transition_matrix(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    P = msm.transition_matrix
+    # should be ndarray by default
+    # assert (isinstance(P, np.ndarray))
+    assert_(isinstance(P, np.ndarray) or isinstance(P, scipy.sparse.csr_matrix))
+    # shape
+    assert_equal(P.shape, (msm.n_states, msm.n_states))
+    # test transition matrix properties
+    import msmtools.analysis as msmana
+
+    assert_(msmana.is_transition_matrix(P))
+    assert_(msmana.is_connected(P))
+    # REVERSIBLE
+    if msm.reversible:
+        assert_(msmana.is_reversible(P))
+
+
+def test_selected_count_fraction(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    # should always be a fraction
+    assert_(0.0 <= msm.count_model.selected_count_fraction <= 1.0)
+    # special case for this data set:
+    assert_equal(msm.count_model.selected_count_fraction, scenario.selected_count_fraction)
+
+
+def test_selected_state_fraction(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    # should always be a fraction
+    assert_(0.0 <= msm.count_model.selected_state_fraction <= 1.0)
+
+
+def test_statdist(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    mu = msm.stationary_distribution
+    # should strictly positive (irreversibility)
+    assert_(np.all(mu > 0))
+    # should sum to one
+    assert_almost_equal(np.sum(mu), 1., decimal=10)
+
+    if est.stationary_distribution_constraint is not None:
+        assert_array_almost_equal(msm.stationary_distribution,
+                                  scenario.stationary_distribution[msm.count_model.state_symbols])
+
+
+def test_eigenvalues(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    # use n_states-2 because sparse eigenvalue problem can only be solved by scipy for k < N-1
+    ev = msm.eigenvalues(msm.n_states - 2)
+    # stochasticity
+    assert_(np.max(np.abs(ev)) <= 1 + 1e-12)
+    # irreducible
+    assert_(np.max(np.abs(ev[1:])) < 1)
+    # ordered?
+    evabs = np.abs(ev)
+    for i in range(0, len(evabs) - 1):
+        assert_(evabs[i] >= evabs[i + 1])
+    # REVERSIBLE:
+    if msm.reversible:
+        assert_(np.all(np.isreal(ev)))
+
+
+def test_eigenvectors(double_well_msm_all):
+    scenario, est, msm = double_well_msm_all
+    if not msm.sparse:
+        k = msm.n_states
```

Comment:
you may change the variables below but not is fine too these are like borderline acceptable